### PR TITLE
feat: support force reset template versions

### PR DESCRIPTION
**Pull request by alexcodelf**
**Problem:**

Can not force refresh template or catalog template version with tag updated.

**Solution:**

Make update template version as default behavior. If user hard reset tag, it will update template version.

**Related Issue:**
#1597

### CODE REVIEW

File: pkg/templates/git.go [hunk 1]
```diff
@@ -74,29 +45,17 @@ func CreateTemplateVersionsFromRepo(
...
-		).
+	return mc.TemplateVersions().CreateBulk(templateVersionCreates...).
+		OnConflict(conflictOptions...).
+		UpdateNewValues().
 		Exec(ctx)
-
-	return err
 }
 
 // SyncTemplateFromRef clones a git repository create template and template version with reference.
```
Comment (thxCode) [inline pkg/templates/git.go:61]: syncTemplateFromRef

File: pkg/templates/git.go [hunk 2]
```diff
@@ -210,23 +192,7 @@ func syncTemplateFromRef(
 		)
 	}
```
Comment (thxCode) [inline pkg/templates/git.go:193]:
```
if entity.ProjectID == "" {
  return ...
}

return ...
```

File: pkg/templates/git.go [hunk 3]
```diff
@@ -186,6 +145,29 @@ func syncTemplateFromRef(
...
+			Schema:     *schema,
+			ProjectID:  entity.ProjectID,
+		}).
+		OnConflict(conflictOptions...).
+		UpdateNewValues().
+		Exec(ctx)
+}
+
+// templateVersionConflictOptions returns template version conflict options with template project id.
+func templateVersionConflictOptions(entity *model.Template) []sql.ConflictOption {
```
Comment (thxCode) [inline pkg/templates/git.go:170]: getTemplateVersionUpsertConfilctOptions

**Review by thxCode — APPROVED** [review]
LGTM

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**